index.js

```diff
@@ -0,0 +1,120 @@
+  'View Employees By Department': { className: 'Employee', method: 'getEmployeesByDeparment', callBack: infoCallBack },
+  'View Employees By Manager': { className: 'Employee', method: 'getEmployeesByManager', callBack: infoCallBack },
+  'Add Employee': { className: 'Employee', method: 'add', callBack: putCallBack },
+  'Update Employee Role': { className: 'Employee', method: 'updateRole' , callBack: putCallBack },
+  'View All Roles': { className: 'Role', method: 'getInfo', callBack: infoCallBack },
+  'Add Role': { className: 'Role', method: 'add', callBack: putCallBack },
+  'View All Departments': { className: 'Department', method: 'getInfo', callBack: infoCallBack },
+  'Add Department': { className: 'Department', method: 'add', callBack: putCallBack },
+  'Delete Employee': { className: 'Employee', method: 'delete', callBack: putCallBack },
+  'Delete Role': { className: 'Role', method: 'delete', callBack: putCallBack },
+  'Delete Department': { className: 'Department', method: 'delete', callBack: putCallBack },
+  'Quit': {}
+};
+
+const QUESTIONS = [
+  {
+    type: "list",
+    name: "request",
+    message: "What would you like to do?",
+    choices: Object.keys(MAIN_MENU)
+  },
+  {
+    type: "confirm",
+    name: "confirm_quit",
+    message: "Are you sure you want to quit?",
+    when: (answers) => answers.request == "Quit",
+  }
+];
+
+
+const main = () => {
+  displayWelcome();
+  connect().then((conn) => {
+    console.log("Hello!\n");
+    db = conn;
+    runMainMenu();
+  });
+};
+
+const displayWelcome = () => {
+  console.log("-----------------------------------------------------------------")
+  console.log(figlet.textSync('Employee\nManager\n', { horizontalLayout: 'full' }));
+  console.log("-----------------------------------------------------------------")
+  console.log("-------------------copyright (c) 1983 IBM------------------------")
+  console.log("\n")
+}
+
+function infoCallBack(methodReturn) {
+  const { promise } = methodReturn;
+  promise.then(([rows]) => {
+    console.table(rows);
+    runMainMenu();
+  }).catch((e) => {
+    console.log(e);
+    runMainMenu();
+  });
+}
+
+function putCallBack(methodReturn) {
+  const { promise, targetClass, message } = methodReturn;
+  if (!promise) runMainMenu();
+
+  promise.then(info => {
+    console.log(`\n${message}\n`);
+    if (!targetClass) runMainMenu();
+
+    targetClass.getInfo().promise.then(([rows]) => {
+      console.table(rows);
+      runMainMenu();
+    }).catch((e) => {
+      console.log(e);
+      runMainMenu();
+    });
+  }).catch((e) => {
+    console.log(e);
+    runMainMenu();
+  });
+}
+
+const runMainMenu = async () => {
+  const responses = await inquirer.prompt(QUESTIONS)
+  const { request, confirm_quit } = responses;
+  if (confirm_quit) {
+    console.log("Good Bye!");
+    db.end();
+    return;
+  } else {
+    const methodReturn = await fulfillRequest(request);
```

Comment:
nice job modularising your code and finding common logic that applies to several operations. I really like the modularity you have given this assignment